File: backend/app/schemas/listing.py
import uuid
from pydantic import BaseModel


class ListingOut(BaseModel):
    id: uuid.UUID
    title: str
    description: str
    city: str
    country: str
    region: str
    trip_style: str
    price_per_night: float
    rating: float
    image_url: str
    latitude: float
    longitude: float
    score: float | None = None

    class Config:
        from_attributes = True

File: frontend/src/types/listing.ts
import type { TripStyle } from "../lib/tripStyles";

export interface Listing {
    id: string;
    title: string;
    description: string;
    city: string;
    country: string;
    region: string;
    trip_style: TripStyle;
    price_per_night: number;
    rating: number;
    image_url: string;
    latitude: number;
    longitude: number;
    score?: number;
}